Add /** */ to cpp/language-configurations.json

sean-mcmanus opened the pull request:
Fixes https://github.com/microsoft/vscode-cpptools/issues/12249 .

This was removed in https://github.com/microsoft/vscode/commit/98fa77a679fbd1095d05c0fcb381d6e9b775d743 .

Then PR https://github.com/microsoft/vscode/pull/160357 added /* */ autoClosingPair.

<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://github.com/microsoft/vscode/wiki/How-to-Contribute#pull-requests
* Associate an issue with the Pull Request.
* Ensure that the code is up-to-date with the `main` branch.
* Include a description of the proposed changes and how to test them.
-->

sean-mcmanus commented:
@bobbrow @Colengms Is there any issue with this?

sean-mcmanus commented:
@bobbrow I talked this over with Colen -- this causes users with setting `"C_Cpp.doxygen.generatedStyle": "/**"` to not get onType doxygen generation. Is that okay or what?

bobbrow commented:
> @bobbrow I talked this over with Colen -- this causes users with setting `"C_Cpp.doxygen.generatedStyle": "/**"` to not get onType doxygen generation. Is that okay or what?

If the user has that setting set, can we replace the language configuration with one that doesn't specify `/**` and make it work?

sean-mcmanus commented:
> If the user has that setting set, can we replace the language configuration with one that doesn't specify `/**` and make it work?

@bobbrow We can? How? Even if users have the generationStyle set to `/**`, they'd still get the buggy indentation for comments that are not before a declaration which auto-generates doxygen, i.e. in a function.

sean-mcmanus commented:
@bobbrow I have a fix for Doxygen comment generation that handles the `/** */` case that results from this change.

sean-mcmanus commented:
@alexr00 What's the timeline on getting this reviewed?

alexr00 commented:
@sean-mcmanus can you walk me through what has changed since https://github.com/microsoft/vscode/issues/77008#issuecomment-511925435 that makes this change better now?

sean-mcmanus commented:
> @sean-mcmanus can you walk me through what has changed since [#77008 (comment)](https://github.com/microsoft/vscode/issues/77008#issuecomment-511925435) that makes this change better now?

@alexr00 Yeah, VS Code added the editor.autoClosingComments setting. Also, we fixed an issue with our Doxygen generation code for our next release. So, our team isn't aware of any downside to this change.

sean-mcmanus commented:
> > @sean-mcmanus can you walk me through what has changed since [#77008 (comment)](https://github.com/microsoft/vscode/issues/77008#issuecomment-511925435) that makes this change better now?
> 
> @alexr00 Yeah, VS Code added the editor.autoClosingComments setting. Also, we fixed an issue with our Doxygen generation code for our next release. So, our team isn't aware of any downside to this change.

Also, after that comment was made, we added the /* */ completion: see https://github.com/microsoft/vscode/pull/160357 .

alexr00 commented on extensions/cpp/language-configuration.json:
```diff
@@ -14,7 +14,8 @@
 		{ "open": "\"", "close": "\"", "notIn": ["string"] },
-		{ "open": "/*", "close": "*/", "notIn": ["string", "comment"] }
+		{ "open": "/*", "close": "*/", "notIn": ["string", "comment"] },
+		{ "open": "/**", "close": " */", "notIn": ["string"] }
```
Should this also be `"notIn": ["string", "comment"]` and not just `"notIn": ["string"]` to match the `/*` rule?

sean-mcmanus replied:
@alexr00 Yes, it needs to be `"notIn": ["string"]` , because when the user types the initial `/*` they're already in a comment, so it doesn't have any effect if `["string", "comment"]` is used -- this matches what is used by TypeScript for the exact same completion of `/**` to ` */`.

alexr00 replied:
Thanks for the explanation!

alexr00 reviewed (approved):
Looks good, thank you!
